### DISCUSSION TITLE
Installation Error on WSL: System has not been booted with systemd as init system (PID 1)

### DISCUSSION THREAD

**Discussion by benhovinga** [coder/code-server#2714] [Q&A]
Installing code-server on WSL2 so I can use my more powerful desktop computer to do dev work while on the road or when I cannot use my personal laptop (ie. at work).

I ran `curl -fsSL https://code-server.dev/install.sh | sh` to install and then tried to run `sudo systemctl enable --now code-server@$USER` to have it auto boot. But when I do I get this error
```bash
Created symlink /etc/systemd/system/default.target.wants/code-server@ben.service → /lib/systemd/system/code-server@.service.
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
```

Maybe I don't fully understand how WSL works or maybe there is another command I could use. I did some googleing and some posts recommend using `service` for other applications running on WSL but I don't think that will work for code-server.

How can I make this work? or do I need to?

- Web Browser: N/A
- Local OS: Windows 10
- Remote OS: Ubuntu (WSL 2) on Windows 10
- Remote Architecture: amd64
- `code-server --version`: 3.8.1

PS: running `code-server` works great and I am able to access it in my browser.

**Comment by code-asher** [comment]
There are some ideas here which I think might help: https://github.com/cdr/code-server/issues/1868

I also found this but I can't vouch for it: https://gist.github.com/djfdyuruiry/6720faa3f9fc59bfdf6284ee1f41f950

I think we output the `systemctl enable` command even for WSL right? It would probably make sense for us to check for WSL in the install script and output a different command that will work.
